packages/Ripple/src/index.js

```diff
@@ -0,0 +1,194 @@
+  }
+
+  @bind
+  startRemoval(id) {
+    const { ripples } = this.state
+    const { opacityTiming } = this.props
+
+    if(!ripples[id].animationEnded || !ripples[id].removable)
+      return false
+
+    // If we can remove it we first start
+    // By decreasing the opacity to 0 and
+    // Then after the animation finishes we
+    // Remove the node
+    const _styles = {
+      ...ripples[id].styles,
+      opacity: '0'
+    }
+
+    ripples[id].styles = _styles
+    this.setState({ ripples })
+
+    setTimeout(e => this.removeRipple(id), opacityTiming || 300)
+  }
+
+  @bind
+  removeRipple(id) {
+    const { ripples } = this.state
+    const { opacityTiming } = this.props
+    const _styles = {
+      ...ripples[id].styles,
+      opacity: '0'
+    }
+    ripples[id].styles = _styles
+    this.setState({ ripples })
+    setTimeout(() => {
+      ripples[id] = undefined
+      this.setState({ ripples })
+    }, 10000)
+  }
+
+  @bind
+  animationEnded(id) {
+    const { ripples } = this.state
+
+    ripples[id].animationEnded = true
+    ripples.removable = true
+    this.setState({ ripples })
+
+    if (ripples[id].removable) {
+      this.removeRipple
+    }
+
+    // This will just work if the user has
+    // fired the mouseup event
+    this.startRemoval(id)
+  }
+
+  @bind
+  startScaling(id) {
+    const { ripples } = this.state
+    const { scaleTiming } = this.props
+    const self = this
+
+    setTimeout(e => this.animationEnded(id), scaleTiming || 400)
+
+    const _styles = {
+      ...ripples[id].styles,
+      transform: 'scale(50)'
+    }
+
+    ripples[id].styles = _styles
+    setTimeout( () => self.setState({ ripples }), 1)
+  }
+
+  @bind
+  createNewRipple(x, y) {
+    const id               = this.state.ripples.length
+    const self             = this
+    const { parent }       = this
+    const { rippleStyles } = this.getStyles()
+    const {
+      ripples,
+      timeouts
+    } = this.state
+
+    const ripple = {
+      animationEnded: false,
+      removable: false,
+      styles: {
+        ...rippleStyles,
+        top: y - (rippleStyles.height / 2),
+        left: x - (rippleStyles.width / 2)
+      }
+    }
+
+    ripples.push(ripple)
+    this.setState({ ripples })
+
+    this.createListener(id)
+    this.startScaling(id)
+  }
+
+  @bind
+  handleMouseDown(event) {
+    event.stopPropagation()
+    const { offsetX, offsetY } = event
+
+    this.createNewRipple(offsetX, offsetY)
+  }
+
+  @bind
+  getStyles() {
+    const {
+      startingSize,
+      scaleTiming,
+      opacityTiming,
+      easing,
+      opacity
+    } = this.props
+
+    const styles = {
+      zIndex: -1,
+      position: 'absolute',
+      height: '100%',
+      width: '100%',
+      overflow: 'hidden',
+      background: 'red',
+
+      rippleStyles: {
+        zIndex: -1,
+        position: 'absolute',
+        height: startingSize || 10,
+        width: startingSize || 10,
+        background: '#9E9E9E',
```

Comment:
Background color should be made customizable for lighter/darker elements or themes.